packages/api/src/external/fhir/document/draft-update-document-reference.ts

```diff
@@ -0,0 +1,73 @@
+import { Attachment, DocumentReference, DocumentReferenceContent } from "@medplum/fhirtypes";
+import { toFHIRSubject } from "../patient";
+import BadRequestError from "../../../errors/bad-request";
+import { OrganizationModel } from "../../../models/medical/organization";
+import { metriportDataSourceExtension } from "../shared/extensions/metriport";
+import { Config } from "../../../shared/config";
+import { S3Utils } from "@metriport/core/external/aws/s3";
+import { appendIdentifierOID, toFHIR } from "../organization";
+import { TEMPORARY } from "../../../shared/constants";
+
+const region = Config.getAWSRegion();
+const s3Utils = new S3Utils(region);
+
+export function docRefCheck(docRef: DocumentReference) {
+  if (!docRef.description) throw new BadRequestError(`Document Reference must have a description`);
+  if (!docRef.type) throw new BadRequestError(`Document Reference must have a type`);
+  if (!docRef.context) throw new BadRequestError(`Document Reference must have context`);
+}
+
+/**
+ *
+ * @param prelimDocRef
+ * @param organization
+ * @param patientId
+ * @param docRefId
+ * @param s3Key
+ * @param bucketName
+ * @returns
+ */
+export function composeDocumentReference(
+  prelimDocRef: DocumentReference,
+  organization: OrganizationModel,
+  patientId: string,
+  docRefId: string,
+  s3Key: string,
+  bucketName: string
+): DocumentReference {
+  const containedOrg = toFHIR(organization);
+  const containedOrgWithOID = appendIdentifierOID(organization, containedOrg);
+  const temporaryExtension = metriportDataSourceExtension;
+  temporaryExtension.valueCoding.code = TEMPORARY;
+
+  const subject = toFHIRSubject(patientId);
+
+  const attachment: Attachment = {
+    title: "pre-upload placeholder",
```

Comment:
When is this updated?

Reply:
Oh, it gets replaced [here](https://github.com/metriport/metriport/pull/984/files#diff-3d42b174b9456e6cba24c22730804f621f249c1013c2aedac4adc1c83c3635daR167), right?

Reply:
Correct, ya